datamodel/samples.d/generate_timestampdefinitions.py

```diff
@@ -0,0 +1,1188 @@
+        need_event_location_for=ALL_EVENT_CLASSIFIER_CODES,
+    )
+
+    # EOSP UC 34
+    generate_special_timestamp(
+        'ESOP',
+        PUBLISHER_PATTERN_CA2ATH,
+        'ARRI',
+        'BRTH',
+        'INBD',
+        'Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution',
+        'jit1_1',
+        need_vessel_position=True,  # FIXME: IFS says Optional, we do not support "required" or "absent".
+    )
+
+    # ATS-Pilotage (Inbound) UC 36 + 40
+    xty_service_timestamps(
+        ACT,
+        ['PILO'],
+        [NULL_VALUE, 'INBD'],
+        "Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution",
+        'jit1_0',
+        include_phase_in_name=True,
+        is_cancelable=True,
+        need_vessel_position_for=ACT,
+        need_event_location_for=ACT,
+    )
+
+    # ATS-Towage (Inbound) UC 37 + 38
+    xty_service_timestamps(
+        ACT,
+        ['TOWG'],
+        [NULL_VALUE, 'INBD'],
+        "Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution",
+        'jit1_0',
+        include_phase_in_name=True,
+        is_cancelable=True,
+        need_vessel_position_for=ACT,
+        need_event_location_for=ACT,
+    )
+
+    # ATS-Mooring (Inbound) UC 39 + 44
+    xty_service_timestamps(
+        ACT,
+        ['MOOR'],
+        [NULL_VALUE, 'INBD'],
+        "Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution",
+        'jit1_0',
+        include_phase_in_name=True,
+        is_cancelable=True,
+        need_event_location_for=ACT,
+    )
+
+    #ATA-Berth UC 41
+    generate_special_timestamp(
+        'ATA-Berth',
+        as_publisher_patterns(['TR'], ['ATH']),
+        'ARRI',
+        'BRTH',
+        'ALGS',
+        "Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution",
+        'jit1_0',
+        need_vessel_position=False,
+    )
+
+    # AT All fast UC 42
+    generate_special_timestamp(
+        'AT All fast',
+        PUBLISHER_PATTERN_CA2ATH,
+        'ARRI',
+        'BRTH',
+        'ALGS',
+        "Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution",
+        'jit1_0',
+        'FAST',
+        need_vessel_position=False,
+    )
+
+    # Gangway Down and Safe UC 43
+    generate_special_timestamp(
+        'Gangway Down and Safe',
+        PUBLISHER_PATTERN_CA2ATH,
+        'ARRI',
+        'BRTH',
+        'ALGS',
+        "Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution",
+        'jit1_0',
+        'GWAY',
+        need_vessel_position=False,
+    )
+
+    # Vessel Readiness for Cargo ops UC 45
+    generate_special_timestamp(
+        'Vessel Readiness for Cargo operations',
+        PUBLISHER_PATTERN_CA2ATH,
+        'ARRI',
+        'BRTH',
+        'ALGS',
+        "Start Cargo Operations And Services",
+        'jit1_0',
+        'VRDY',
+        need_vessel_position=False,
+        need_event_location=True,
+    )
```

Comment:
Definitely not a JIT 1.0 timestamp.  Also, I believe there is a JIT 1.1 variant of this relying on `SAFE` rather than `VRDY` (so we need this one twice).

Reply:
added the safe variant and adjusted to the correct JIT version